src/utils/normalizeModuleFederationOptions.ts

```diff
@@ -133,9 +133,17 @@ function searchPackageVersion(sharedName: string): string | undefined {
       path.parse(potentialPackageJsonDir).base !== 'node_modules' &&
       potentialPackageJsonDir !== rootDir
     ) {
-      const potentialPackageJson = path.join(potentialPackageJsonDir, 'package.json');
-      if (fs.existsSync(potentialPackageJson)) {
-        return require(potentialPackageJson).version;
+      const potentialPackageJsonPath = path.join(potentialPackageJsonDir, 'package.json');
+      if (fs.existsSync(potentialPackageJsonPath)) {
+        const potentialPackageJson = require(potentialPackageJsonPath);
+        if (
+          typeof potentialPackageJson == 'object' &&
+          potentialPackageJson !== null &&
+          typeof potentialPackageJson.version === 'string' &&
+          potentialPackageJson.name === sharedName
+        ) {
```

Comment:
Adding additional checks for the package.json candidate:
- is an object
- has a `version` property that's a string
- has a `name` property that matches what we're looking for

Only then do we return the version. Otherwise, we continue looking up the directory tree.

Reply:
Thanks for the explanation